### PR TITLE
first version

### PR DESCRIPTION
**Pull request by IegorSoboliev**

### CODE REVIEW

**Comment by IegorSoboliev** [comment]
Please, don`t review first version. Start with Done

File: src/main/java/mate/academy/internet/shop/dao/impl/BucketDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import java.util.Optional;
+
+import mate.academy.internet.shop.dao.BucketDao;
+import mate.academy.internet.shop.database.Storage;
+import mate.academy.internet.shop.lib.Dao;
+import mate.academy.internet.shop.model.Bucket;
+
+@Dao
+public class BucketDaoImpl implements BucketDao {
+    private static Long bucketCounter = 1L;
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/dao/impl/BucketDaoImpl.java]:
```suggestion
    private static Long idGenerator = 1L;
```

File: src/main/java/mate/academy/internet/shop/dao/impl/BucketDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    @Override
+    public Bucket update(Bucket bucket) {
+        for (Bucket b : Storage.buckets) {
+            if (b.getId().equals(bucket.getId())) {
+                b.setSelectedItems(bucket.getSelectedItems());
+            }
+        }
+        return bucket;
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/dao/impl/BucketDaoImpl.java]: try to use Stream api

File: src/main/java/mate/academy/internet/shop/dao/impl/BucketDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    @Override
+    public boolean deleteById(Long id) {
+        for (Bucket b : Storage.buckets) {
+            if (b.getId().equals(id)) {
+                Storage.buckets.remove(b);
+                return true;
+            }
+        }
+        return false;
+    }
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/dao/impl/BucketDaoImpl.java]: try to use removeIf method

File: src/main/java/mate/academy/internet/shop/dao/impl/ItemDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import java.util.Optional;
+
+import mate.academy.internet.shop.dao.ItemDao;
+import mate.academy.internet.shop.database.Storage;
+import mate.academy.internet.shop.lib.Dao;
+import mate.academy.internet.shop.model.Item;
+
+@Dao
+public class ItemDaoImpl implements ItemDao {
+    private static Long itemCounter = 1L;
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/dao/impl/ItemDaoImpl.java]: idGenerator

File: src/main/java/mate/academy/internet/shop/dao/impl/ItemDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+    @Override
+    public Item update(Item item) {
+        for (Item i : Storage.items) {
+            if (i.getId().equals(item.getId())) {
+                i.setName(i.getName());
+                i.setPrice(i.getPrice());
+            }
+        }
+        return item;
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/dao/impl/ItemDaoImpl.java]: Stream API

File: src/main/java/mate/academy/internet/shop/dao/impl/OrderDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+import java.util.List;
+import java.util.Optional;
+
+import mate.academy.internet.shop.database.Storage;
+import mate.academy.internet.shop.lib.Dao;
+import mate.academy.internet.shop.model.Order;
+
+@Dao
+public class OrderDaoImpl implements mate.academy.internet.shop.dao.OrderDao {
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/dao/impl/OrderDaoImpl.java]:
```suggestion
public class OrderDaoImpl implements OrderDao {
```
треба, щоб так виходило :)

File: src/main/java/mate/academy/internet/shop/dao/impl/OrderDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    private static Long orderCounter = 1L;
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/dao/impl/OrderDaoImpl.java]: idGenerator

File: src/main/java/mate/academy/internet/shop/dao/impl/OrderDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    public Order create(Order order) {
+        order.setId(orderCounter);
+        Storage.orders.add(order);
+        orderCounter++;
+        return order;
+    }
+
+    @Override
+    public Order update(Order order) {
+        for (Order o : Storage.orders) {
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/dao/impl/OrderDaoImpl.java]: Stream api

File: src/main/java/mate/academy/internet/shop/model/Bucket.java [hunk 9]
```diff
@@ -0,0 +1,46 @@
+package mate.academy.internet.shop.model;
+
+import java.util.List;
+
+public class Bucket {
+    private Long id;
+    private User user;
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/model/Bucket.java]:
```suggestion
    private Long userId;
```

File: src/main/java/mate/academy/internet/shop/model/Bucket.java [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+    private List<Item> selectedItems;
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/model/Bucket.java]:
```suggestion
    private List<Item> items;
```

File: src/main/java/mate/academy/internet/shop/model/Order.java [hunk 11]
```diff
@@ -0,0 +1,60 @@
+package mate.academy.internet.shop.model;
+
+import java.util.List;
+
+public class Order {
+    private Long id;
+    private User user;
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/model/Order.java]:
```suggestion
    private Long userId;
```

File: src/main/java/mate/academy/internet/shop/model/Order.java [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+    private List<Item> items;
+    private Bucket bucket;
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/model/Order.java]: Bucket is not needed at Order. please, remove this field

File: src/main/java/mate/academy/internet/shop/service/BucketService.java [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+import mate.academy.internet.shop.model.Bucket;
+import mate.academy.internet.shop.model.Item;
+
+public interface BucketService {
+
+    Bucket create(Bucket bucket);
+
+    Bucket update(Bucket bucket);
+
+    Bucket get(Long id);
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/service/BucketService.java:14]: в сервісах зроби Optional, як в дао

File: src/main/java/mate/academy/internet/shop/dao/impl/BucketDaoImpl.java (hunk 2 above)
Comment (IegorSoboliev) [reply src/main/java/mate/academy/internet/shop/dao/impl/BucketDaoImpl.java]: You mean?         return Storage.buckets
                .stream()
                .filter(b -> b.getId().equals(bucket.getId()))
                .findFirst()
                .ifPresent(b -> b.setSelectedItems(bucket.getSelectedItems());

File: src/main/java/mate/academy/internet/shop/dao/impl/BucketDaoImpl.java (hunk 3 above)
Comment (IegorSoboliev) [reply src/main/java/mate/academy/internet/shop/dao/impl/BucketDaoImpl.java]: You mean?         return Storage.buckets.removeIf(b -> b.getId().equals(id));

File: src/main/java/mate/academy/internet/shop/service/BucketService.java (hunk 13 above)
Comment (IegorSoboliev) [reply src/main/java/mate/academy/internet/shop/service/BucketService.java:14]: Had opposite recommendation

File: src/main/java/mate/academy/internet/shop/dao/impl/BucketDaoImpl.java (hunk 1 above)
Comment (IegorSoboliev) [reply src/main/java/mate/academy/internet/shop/dao/impl/BucketDaoImpl.java]: Ok

File: src/main/java/mate/academy/internet/shop/dao/impl/ItemDaoImpl.java (hunk 4 above)
Comment (IegorSoboliev) [reply src/main/java/mate/academy/internet/shop/dao/impl/ItemDaoImpl.java]: Ok

File: src/main/java/mate/academy/internet/shop/model/Bucket.java (hunk 10 above)
Comment (IegorSoboliev) [reply src/main/java/mate/academy/internet/shop/model/Bucket.java]: Ok

File: src/main/java/mate/academy/internet/shop/dao/impl/BucketDaoImpl.java (hunk 2 above)
Comment (IegorSoboliev) [reply src/main/java/mate/academy/internet/shop/dao/impl/BucketDaoImpl.java]: Or ?         Storage.buckets
                .stream()
                .filter(b -> b.getId().equals(bucket.getId()))
                .findFirst()
                .ifPresent(b -> b.setSelectedItems(bucket.getSelectedItems()));
        return bucket;

File: src/main/java/mate/academy/internet/shop/dao/impl/OrderDaoImpl.java (hunk 6 above)
Comment (IegorSoboliev) [reply src/main/java/mate/academy/internet/shop/dao/impl/OrderDaoImpl.java]: Sorry)

File: src/main/java/mate/academy/internet/shop/model/Bucket.java (hunk 9 above)
Comment (IegorSoboliev) [reply src/main/java/mate/academy/internet/shop/model/Bucket.java]: ok

File: src/main/java/mate/academy/internet/shop/model/Order.java (hunk 12 above)
Comment (IegorSoboliev) [reply src/main/java/mate/academy/internet/shop/model/Order.java]: Have a question

File: src/main/java/mate/academy/internet/shop/model/Order.java (hunk 11 above)
Comment (IegorSoboliev) [reply src/main/java/mate/academy/internet/shop/model/Order.java]: Done

File: src/main/java/mate/academy/internet/shop/dao/impl/ItemDaoImpl.java (hunk 5 above)
Comment (IegorSoboliev) [reply src/main/java/mate/academy/internet/shop/dao/impl/ItemDaoImpl.java]: Done

File: src/main/java/mate/academy/internet/shop/dao/impl/BucketDaoImpl.java (hunk 3 above)
Comment (IegorSoboliev) [reply src/main/java/mate/academy/internet/shop/dao/impl/BucketDaoImpl.java]: Done

File: src/main/java/mate/academy/internet/shop/dao/impl/BucketDaoImpl.java (hunk 2 above)
Comment (IegorSoboliev) [reply src/main/java/mate/academy/internet/shop/dao/impl/BucketDaoImpl.java]: Done elsewhere

File: src/main/java/mate/academy/internet/shop/model/User.java [hunk 14]
```diff
@@ -0,0 +1,45 @@
+package mate.academy.internet.shop.model;
+
+public class User {
+    private java.lang.Long id;
```
Comment (Tarabarnik) [inline src/main/java/mate/academy/internet/shop/model/User.java:4]:
```suggestion
    private Long id;
```

**Review by Tarabarnik — APPROVED** [review]

**Pull request merged**